<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="exporter-version" content="Evernote Mac 6.6.1 (453373)"/><meta name="keywords" content="cervus, error, Parentage, Pinsky"/><meta name="altitude" content="26.79477119445801"/><meta name="author" content="Michelle Stuart"/><meta name="created" content="2016-01-04 19:16:59 +0000"/><meta name="latitude" content="40.47723495214617"/><meta name="longitude" content="-74.43835649500053"/><meta name="source" content="desktop.mac"/><meta name="updated" content="2016-05-09 03:06:19 +0000"/><title>Notes on Cervus Parentage analysis</title></head><body>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div>
<hr/></div>
<div><span style="color: rgb(148, 33, 147);">January 4, 2016</span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);">The gene pop file created by rad_haplotyper is a 3 digit file, but it seems to have some bugs in it.</span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);">I most recently used vcf2genepop.pl to create the gene pop file from the SNP.xxx.HWE.recode.vcf file.  </span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><b><span style="color: rgb(148, 33, 147);">Parentage output analysis:</span></b></div>
<ul>
<li><span style="color: rgb(148, 33, 147);">asterisk is a strict parent assignment - focus on these for parent pairs (the trio column)</span></li>
<li><span style="color: rgb(148, 33, 147);">plus sign is a relaxed parent assignment</span>
<ol>
<li><span style="color: rgb(148, 33, 147);">if there is a star in the trio, pull lat long of both parents and get distances between parents as well as offspring</span></li>
<li><span style="color: rgb(148, 33, 147);">if no star in trio, look for star in either pair confidence and get distance between that parent and offspring</span></li>
<li><span style="color: rgb(148, 33, 147);">run identity and eliminate individuals that have 2 APCL ID’s in the parent/offspring files (???) - this is from the note <a style="color: rgb(105, 170, 53);" href="evernote:///view/154400311/s5/f7ea898c-4137-46ef-b27e-0ce820fc6316/f7ea898c-4137-46ef-b27e-0ce820fc6316/"><span style="color: rgb(148, 33, 147);">2015-05-14 Thursday</span></a></span></li>
<li><span style="color: rgb(148, 33, 147);">filter by ID and count the number of offspring</span></li>
</ol>
</li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div>
<hr/></div>
<div><span style="color: rgb(148, 33, 147);">February 3, 2016 </span></div>
<div><span style="color: rgb(148, 33, 147);">Run parentage with the fixed Parent Sim files created on Jan 22/23.  Those files were created using the DP20_edited </span><font color="#942193">gene pop file - same as original dDocent filtered gene pop minus the APOC and APPE samples.  The DP20edited_AF allele frequency file was used as input, that file can be found in the DP20_07-JAN-16 directory, used the cervus project file saved in the Jan 07 directory to get started...</font></div>
<ul>
<li><font color="#942193">open the cervus project DP20_edited.crv found in DP20_07-JAN-16 folder in Michelle’s folder on the desktop</font></li>
<li><span style="color: rgb(148, 33, 147);">click on Analysis &gt; Parentage Analysis &gt; Parent Pair (Sexes Unknown)</span>
<ul>
<li><span style="color: rgb(148, 33, 147);">04_Jan_16_juve.txt is autofilled</span><span style="color: rgb(148, 33, 147);"> </span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">Offspring in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">Does not include candidate parents</span></li>
<li><span style="color: rgb(148, 33, 147);">04_Jan_16_adult.txt is autofilled</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">One column for all offspring</span></li>
<li><span style="color: rgb(148, 33, 147);">Candidate ID in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">Genotype file DP20edited_CONV.csv is autofilled for genotype file</span></li>
<li><span style="color: rgb(148, 33, 147);">There is a header row</span></li>
<li><span style="color: rgb(148, 33, 147);">ID is in column 2</span></li>
<li><span style="color: rgb(148, 33, 147);">First allele is in column 3</span></li>
<li><span style="color: rgb(148, 33, 147);">Allele frequency data DP20edited_AF.alf is autofilled</span></li>
<li><span style="color: rgb(148, 33, 147);">Simulation data is autofilled to DP20edit_PS.sim - <span style="background-color:rgb(255, 250, 165);-evernote-highlight:true;">switch to the new PS file from Jan 22</span></span></li>
<li><span style="background-color:rgb(255, 250, 165);-evernote-highlight:true;"><font color="#942193">Save as new PR files with today’s date DP20_03-FEB-16</font></span></li>
<li><span style="color: rgb(148, 33, 147);">for each offspring include the two most likely parents</span></li>
<li><span style="color: rgb(148, 33, 147);">sort by joint LOD scores</span></li>
<li><span style="background-color:rgb(255, 250, 165);-evernote-highlight:true;"><font color="#FF2600">Did not run!!! changing the plan - have to wait for edited gene pop and genotyping error numbers</font></span></li>
<li><font color="#942193">when finished, save project as a cry in todays directory DP20_03-FEB-16</font></li>
</ul>
</li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);">Also want to try using all fish as juveniles and only adults in the adult file and see what happens</span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div>
<hr/></div>
<div><span style="color: rgb(148, 33, 147);">February 17, 2016 </span></div>
<div><font color="#942193">Run Parent_sim on new gene pop from which regenotyped samples and true recaptures have been removed (6 false positives were left in, while they are very well matched, uncertain where the error is falling, either fish moved too great of a distance to be realistic, or there was a lab or label error somewhere)</font></div>
<div><br/></div>
<ul>
<li><span style="color: rgb(148, 33, 147);"><font face="monospace">Click on analysis&gt;simulation of parentage analysis&gt;sexes of pair unknown</font></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">20k offspring</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">1000 parents</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">prop sampled 0.1</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">prop loci typed 0.75</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">mistyped 0.001</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">min typed loci 30% of total = <b>786</b></span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">use LOD (not Delta)</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">relaxed 80%, strict 95%</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;"><b>options: Distributions tab:</b></span></span></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">LOD distrbution 100 categories, 0-100</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">table of LOD scores</span></span></span></li>
</ul>
</li>
<li><span style="color: rgb(148, 33, 147);"><b><span style="font-size: 12px;"><span style="font-family: Helvetica;">Liklihood tab:</span></span></b></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">use mistyping rate as error rate in calculations</span></span></span></li>
</ul>
</li>
</ul>
</li>
</ul>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);"><br/></span></span></span></div>
<div>
<hr/></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);">February 29, 2016 </span></span></span></div>
<div><font face="Helvetica" color="#942193" size="2">Cervus keeps stalling and won’t complete a parent sim on a DP 10 gene pop file with 881 individuals and 2620 loci.  Malin suggested cutting down the number of SNPs by changing the maf in our filtering to 0.2 instead of 0.05.  Going to try converting the edited geneopop to vcf and then running the maf filter, then converting back to gene pop.  </font><a style="color:#69aa35;" href="evernote:///view/154400311/s5/025f5269-cc44-40d4-98c7-4b692c6b36c7/025f5269-cc44-40d4-98c7-4b692c6b36c7/">Filtering February 29, 2016</a></div>
<div><br/></div>
<div>Converted the genepop</div>
<div>Ran through AF - 1012 loci</div>
<div><br/></div>
<ul>
<li><span style="color: rgb(148, 33, 147);"><font face="monospace">Click on analysis&gt;simulation of parentage analysis&gt;sexes of pair unknown</font></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">20k offspring</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">1000 parents</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">prop sampled 0.1</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">prop loci typed 0.75</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">mistyped 0.001</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">min typed loci 30% of total = <b>303</b></span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">use LOD (not Delta)</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">relaxed 80%, strict 95%</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;"><b>options: Distributions tab:</b></span></span></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">LOD distrbution 100 categories, 0-100</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">table of LOD scores</span></span></span></li>
</ul>
</li>
<li><span style="color: rgb(148, 33, 147);"><b><span style="font-size: 12px;"><span style="font-family: Helvetica;">Liklihood tab:</span></span></b></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">use mistyping rate as error rate in calculations</span></span></span></li>
</ul>
</li>
</ul>
</li>
</ul>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);"><br/></span></span></span></div>
<div>
<hr/></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);">March 4, 2016 </span></span></span></div>
<div><font face="Helvetica" color="#942193" size="2">Running real parent on the DP10 reduced filtered, regenotype and recapture removed gene pop- DP10maf2_29-FEB-2016 - need to use the new adult and juvenile files created using the top two information...</font><a style="color:#69aa35;" href="evernote:///view/154400311/s5/8dd524f1-47f9-4767-bbea-4ded695a15ab/8dd524f1-47f9-4767-bbea-4ded695a15ab/">2016-02-17 Wednesday</a> - need Lightning for this...</div>
<div>Have to rename the parent samples according to the dDocent rosetta stone naming scheme:</div>
<ul>
<li>In R, read in the gene pop - create a sample ID for the gene pop samples, merge with the adult file and drop all that are not adults, drop all genotype info so there are only sample IDs…opened the adults_juvenile code and I did it already, the file is called adults_renamed!!  Yay Michelle!!!</li>
<li><span style="color: rgb(148, 33, 147);">click on Analysis &gt; Parentage Analysis &gt; Parent Pair (Sexes Unknown)</span>
<ul>
<li><span style="color: rgb(148, 33, 147);">select juvenile file</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">Offspring in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">Does not include candidate parents</span></li>
<li><span style="color: rgb(148, 33, 147);">select adult file</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">One column for all offspring</span></li>
<li><span style="color: rgb(148, 33, 147);">Candidate ID in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">select genotype file (converted </span><font color="#942193">gene pop)</font></li>
<li><span style="color: rgb(148, 33, 147);">There is a header row</span></li>
<li><span style="color: rgb(148, 33, 147);">ID is in column 2</span></li>
<li><span style="color: rgb(148, 33, 147);">First allele is in column 3</span></li>
<li><span style="color: rgb(148, 33, 147);">Allele frequency data is autofilled</span></li>
<li><span style="color: rgb(148, 33, 147);">Simulation data is autofilled </span></li>
<li><font color="#942193">Save as new PR files with today’s date </font></li>
<li><span style="color: rgb(148, 33, 147);">for each offspring include the two most likely parents</span></li>
<li><span style="color: rgb(148, 33, 147);">sort by joint LOD scores</span></li>
<li><font color="#942193">when finished, save project as a crv in the directory</font></li>
</ul>
</li>
<li><span style="color: rgb(255, 38, 0);">It made zero assignments, I wonder if this is because there are quotes around all of the adult ID’s.  Remove quotes in sublime, reimport file and try again.</span></li>
<li><font color="#942193">Trying again with same parameters as above, taking longer this time which is a good sign…It made 5 assignments but said there were only 50 offspring, which doesn’t make much sense.  Need to compare the adult, offspring, and converted files to see why it isn’t finding more individuals.  <span style="color: rgb(255, 38, 0);">For some reason there are only a few individuals in our converted file</span>…need to re-run convert, AF, and PS.  Argh.</font></li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><font color="#942193">Examining the procedure used to create the current gene pop file</font></div>
<ul>
<li><span style="color: rgb(148, 33, 147);">Filtered for a new maf using vcftools - according to note, had 990 individuals at that time (good) </span><a style="color:#69aa35;" href="evernote:///view/154400311/s5/025f5269-cc44-40d4-98c7-4b692c6b36c7/025f5269-cc44-40d4-98c7-4b692c6b36c7/">Filtering February 29, 2016</a></li>
<li><span style="color: rgb(148, 33, 147);">Ran through rosetta stone - output file is 2016-02-29_renamed.genepop - has 990 individuals</span></li>
<li><font color="#942193">ran through regenotypes - output file is noregenos.genepop - has 97 individuals…rechecking here</font></li>
<li><span style="color: rgb(148, 33, 147);">ran through recaptures</span></li>
<li><span style="color: rgb(148, 33, 147);">ran through cervus</span></li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div>Opening regeno_evaled.R</div>
<div>Looks like I used the wrong gene pop file when I ran this last time...</div>
<div>Also I changed the way we keep rows and the new way doesn’t keep everything. Have to change all of the NA’s to “KEEP” in order for it to work. - 2016-03-04_noregenos.genepop</div>
<div><br/></div>
<div>Rerun this new file through recaptures: 2016-03-04_norecaps.genepop</div>
<div><br/></div>
<div>Rerun this new gene pop through cervus - DP10maf2_04-MAR-16</div>
<ul>
<li><span style="color: rgb(148, 33, 147);"><font face="monospace">Click on analysis&gt;simulation of parentage analysis&gt;sexes of pair unknown</font></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">20k offspring</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">1000 parents</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">prop sampled 0.1</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">prop loci typed 0.75</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">mistyped 0.001</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">min typed loci 30% of total = <b>303</b></span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">use LOD (not Delta)</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">relaxed 80%, strict 95%</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;"><b>options: Distributions tab:</b></span></span></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">LOD distrbution 100 categories, 0-100</span></span></span></li>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">table of LOD scores</span></span></span></li>
</ul>
</li>
<li><span style="color: rgb(148, 33, 147);"><b><span style="font-size: 12px;"><span style="font-family: Helvetica;">Liklihood tab:</span></span></b></span>
<ul>
<li><span style="color: rgb(148, 33, 147);"><span style="font-size: 12px;"><span style="font-family: Helvetica;">use mistyping rate as error rate in calculations</span></span></span></li>
</ul>
</li>
</ul>
</li>
</ul>
<div><font face="Helvetica" color="#942193" size="2">I’m a little worried because it says it will take 4 hours to run, last time it said it would take 23 hours. -</font></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);"><br/></span></span></span></div>
<div>
<hr/></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);">March 5, 2016 </span></span></span></div>
<div><font face="Helvetica" color="#942193" size="2">Still running under no-response going to wait 48 hours and see what happens</font></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);"><br/></span></span></span></div>
<div>
<hr/></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);">March 6, 2016 </span></span></span></div>
<div><span style="font-size: 12px;"><span style="font-family: Helvetica;"><span style="color: rgb(148, 33, 147);">Finished - run parentage</span></span></span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<ul>
<li><span style="color: rgb(148, 33, 147);">click on Analysis &gt; Parentage Analysis &gt; Parent Pair (Sexes Unknown)</span>
<ul>
<li><span style="color: rgb(148, 33, 147);">select juvenile file</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">Offspring in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">Does not include candidate parents</span></li>
<li><span style="color: rgb(148, 33, 147);">select adult file</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">One column for all offspring</span></li>
<li><span style="color: rgb(148, 33, 147);">Candidate ID in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">select genotype file (converted </span><font color="#942193">gene pop)</font></li>
<li><span style="color: rgb(148, 33, 147);">There is a header row</span></li>
<li><span style="color: rgb(148, 33, 147);">ID is in column 2</span></li>
<li><span style="color: rgb(148, 33, 147);">First allele is in column 3</span></li>
<li><span style="color: rgb(148, 33, 147);">Allele frequency data is autofilled</span></li>
<li><span style="color: rgb(148, 33, 147);">Simulation data is autofilled </span></li>
<li><font color="#942193">Save as new PR files with today’s date </font></li>
<li><span style="color: rgb(148, 33, 147);">for each offspring include the two most likely parents</span></li>
<li><span style="color: rgb(148, 33, 147);">sort by joint LOD scores</span></li>
<li><font color="#942193">when finished, save project as a crv in the directory</font></li>
</ul>
</li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div>
<hr/></div>
<div><span style="color: rgb(148, 33, 147);">March 7, 2016 </span></div>
<div><span style="color: rgb(148, 33, 147);">Parentage finished with 817 assignments for parent pairs.  Need to examine parent pairs to see if they are at the same site, same anem.  Going to import the PR csv into mega analysis to examine this.</span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);">Analyzing parentage results:</span></div>
<div><span style="color: rgb(148, 33, 147);">Made a csv and sent to Malin of all parents that are at the same sites and have a strict Trio assignment.  There is a wide range of LOD scores.</span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><font color="#942193">Examining how many individuals were assigned to each parent and with how many reads</font></div>
<ul>
<li><span style="color: rgb(148, 33, 147);">copy first candidate ID over to new sheet</span></li>
<li><span style="color: rgb(148, 33, 147);">count each instance</span></li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div>
<hr/></div>
<div><span style="color: rgb(148, 33, 147);">March 8, 2016 </span></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><span style="color: rgb(148, 33, 147);">Speaking with Malin,</span></div>
<div><span style="color: rgb(148, 33, 147);">Sim looks good for DP10maf2-4MAR2016</span></div>
<div><br/></div>
<div>No obvious problems on the pr summary report other than more assignments than we would expect</div>
<div>Malin talking:</div>
<div><a href="Notes%20on%20Cervus%20Parentage%20analysis.resources/Evernote%2020160308%2013%3A49%3A03.m4a">Evernote 20160308 13:49:03.m4a</a></div>
<div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;">I’ve been cleaning up the trio strict data a bit, looking up sites for fish where no site was automatically generated based on my formula (which is changing some NO’s to YES’s).  </div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;">I noticed an interesting case which may be common among these assignments:</div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;">APCL13_203 is assigned to the parent pair APCL12_095 and APCL14_435.</div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;">APCL12_095 is assigned to the parent pair APCL13_203 and APCL14_435</div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;">
<div>APCL14_435 is assigned to the parent pair APCL12_095 and APCL14_176.</div>
</div>
<div style="color: rgb(0, 0, 0); font-family: 'Helvetica Neue'; font-size: 12px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px;">
<div><br/></div>
</div>
</div>
<div>Are they related?  Are they parents?  Are they recaptures?</div>
<div>How to tease apart this knot.  </div>
<div><br/></div>
<div>Bring in descriptive data, calculate distance between parent pairs and offspring.</div>
<div><br/></div>
<div>Just looking at LOD scores, there is a good chance that APCL13_203 and APCL12_095 are in a parental relationship and APCL14_435 is not the parent of APCL13_203.</div>
<div><br/></div>
<div>APCL14_435 is not a big fish and the other two are (APCL14_176</div>
<div><br/></div>
<div>
<hr/></div>
<div>March 25, 2016</div>
<div>Conversation with Malin:</div>
<div>Limit pool of parents and offspring more - use tail color? - need to make sure we are pulling the right size/tail color for the correct year - <span style="background-color: rgb(255, 250, 165);-evernote-highlight:true;">Malin will work on how to define parents</span></div>
<div><br/></div>
<div>Run by year - 2012 parents and 2013 parents and match to 2013 offspring, 2012, 2013, 2014 parents to run 2014 offspring - relational database</div>
<div><br/></div>
<div>master database of individuals, table of observations, table of genotype data</div>
<div><br/></div>
<div>Colony - Malin got it to run at some point</div>
<div><br/></div>
<div>- Malin needs reports!!! - mislabel</div>
<div>- Malin needs report on identity - new identity</div>
<div>- I need a definition of an Adult from Malin</div>
<div><br/></div>
<div>
<hr/></div>
<div>April 12, 2016 </div>
<div>Malin provided definition of adult and juvenile and we are going to initially create a pool of all juveniles (from all years) and all adults and then run through cervus parentage to see what happens ( had been debating running by year or trying to determine what would happen if a fish changed from J to P from year to year, at this point going forward with large pools because recaptures were eliminated (one of the gene pops had recaptures eliminated).</div>
<div><a style="color:#69aa35;" href="evernote:///view/154400311/s5/98111d9d-375d-45cc-878f-3694a06254e9/98111d9d-375d-45cc-878f-3694a06254e9/">Cervus Parentage April 12, 2016</a></div>
<div><br/></div>
<div>This run failed because sample IDs in our database do not match Sample ID’s in the gene pop, the sample ID’s in the gene pop are in dDocent format (APCL_15203L3423) and the sample ID’s in the database are Pinsky Lab format (APCL15_203).  Need to rename gene pop IDs so that it is useable in our database - write an R script.</div>
<div><br/></div>
<div>
<hr/></div>
<div>April 29, 2016 </div>
<div>Revising the naming issue.  Going to run the rosetta stone script and see if this is the best place to rename the samples.  Currently we know if a sample had an identity issue (lab error or mislabeled sample) because the dDocent ID is followed by an underscore and the proposed correction.  We can retain this information by changing the dDocent ID back to a Pinsky Lab ID and still using the underscore proposed correction - scripts that look at ID can search the last 10 characters of an ID.</div>
<div><br/></div>
<div>Moved the R scripts for post-dDocent processing into the google drive cervus folder so that Malin has access while I’m in the Philippines.  The rosetta_genepop.R file is located in the Google Drive - /Pinsky Lab/Cervus/Michelle's R codes/Genetics/rosetta stone genepop/</div>
<div><br/></div>
<div>Updated script working directory and source, the script is using the genfile  <span style="font: 11.3px Monaco; color: #b01600">'2016_03_21_DP10g95c9maf05.genepop’ </span></div>
<div>This gene pop is for SEQ03-SEQ13, Malin wants to focus on SEQ03-SEQ09 until we get the data from SEQ15,16 back so using the previous gene pop,  <span style="font: 11.3px Monaco; color: #b01600">'DP10g95maf2.genepop’, which was created</span> <a style="color:#69aa35;" href="evernote:///view/154400311/s5/025f5269-cc44-40d4-98c7-4b692c6b36c7/025f5269-cc44-40d4-98c7-4b692c6b36c7/">Filtering February 29, 2016 - SNP.DP10g95maf05.HWE - 1012 loci</a>.</div>
<div><br/></div>
<div>Everything looking good until I look for missing names and it looks like they are all missing.  Need to go into the google sheet to see what is going on.</div>
<div><br/></div>
<div>
<hr/></div>
<div>April 30, 2016</div>
<div>It appears that I changed the rosetta plan somewhat.  Originally, this file was to include all sample names, and now it only includes names that need to be changed due to error.  Need to re-word the R script.</div>
<div><br/></div>
<div>Fixed code through outputting a gene pop that has been renamed and has regenotypes removed - ready to put into Cervus for identity analysis <a style="color:#69aa35;" href="evernote:///view/154400295/s5/daa13888-9199-4505-b933-d73d7d9ecbd3/daa13888-9199-4505-b933-d73d7d9ecbd3/">Notes on Cervus Identity analysis</a></div>
<div><br/></div>
<div><br/></div>
<div>
<hr/></div>
<div>May 1, 2016 </div>
<div>Imported the gene pop that has been renamed, with all regenotyped sample and recaptured fish removed.</div>
<div><br/></div>
<div>Converted noregeno gene pop &amp; named convert2, used  AF file from identity analysis done today. 865 indie, 1012 loci</div>
<div><br/></div>
<div>Needed to create an offspring file with names that match the gene pop, see updated R script. Made a new script in R called adult_juvenile.R that is stored in the code folder of the Identity &amp; Parentage_2016-05_01 folder in the Cervus folder on google drive.</div>
<div><br/></div>
<div>Starting parentage analysis on cervus:</div>
<ul>
<li><span style="color: rgb(148, 33, 147);">select juvenile file</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">Offspring in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">Does not include candidate parents</span></li>
<li><span style="color: rgb(148, 33, 147);">select adult file</span></li>
<li><span style="color: rgb(148, 33, 147);">no header </span></li>
<li><span style="color: rgb(148, 33, 147);">One column for all offspring</span></li>
<li><span style="color: rgb(148, 33, 147);">Candidate ID in column 1</span></li>
<li><span style="color: rgb(148, 33, 147);">select genotype file (converted </span><font color="#942193">gene pop) (convert2)</font></li>
<li><span style="color: rgb(148, 33, 147);">There is a header row</span></li>
<li><span style="color: rgb(148, 33, 147);">ID is in column 2</span></li>
<li><span style="color: rgb(148, 33, 147);">First allele is in column 3</span></li>
<li><span style="color: rgb(148, 33, 147);">Allele frequency data is autofilled</span></li>
<li><span style="background-color: rgb(255, 250, 165);-evernote-highlight:true;"><span style="color: rgb(148, 33, 147);">Simulation data is from March 4th simulation run</span></span></li>
<li><font color="#942193">Save as new PR files with today’s date </font></li>
<li><span style="color: rgb(148, 33, 147);">for each offspring include the two most likely parents</span></li>
<li><span style="color: rgb(148, 33, 147);">sort by joint LOD scores</span></li>
<li><font color="#942193">when finished, save project as a crv in the directory</font></li>
</ul>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div><br/></div>
<div><span style="color: rgb(148, 33, 147);"><br/></span></div>
<div style="margin-left:40px;"><br/></div>
</body></html>