---
title: "Pinsky Lab Cookbook"
output: 
  github_document: 
    toc: TRUE
    pandoc_args: --webtex
  html_notebook:
    toc: TRUE
---

This document exists to share institutional knowledge for obstacles that are commonly run into by Pinsky Lab Members.  The struggle is real!

# Github
## How to add undergrads to the Pinsky Lab github.
1. Make sure the undergrad has signed up for github and note whatever email they used to sign up.
1. Create the repository on github, this should be the project that they are working on.
1. Invite the undergrad as a collaborator.

## How to get RStudio and github to talk to each other
1. Create a repository on github, copy the URL.
1. Create a new project in RStudio and paste in the URL from github

## How to move an existing project on github into RStudio
1. Copy the URL from github for your repository.
1. On your computer, add "_old" to the end of the name of your folder.
1. Create a new project in Rstudio and paste in the URL from github with the normal folder name.
1. Open the "_old" folder, copy all, open the new folder, paste all.  

## How to transfer ownership of a repository from personal to pinskylab in github
1. Click on the repository in github
1. Click on "settings" in the upper middle right.
1. Scroll down to click on "transfer ownership".

## How to connect to one script in someone else's github repo
1. Go to the webpage that contains the script.  
1. Click on RAW and then copy the URL.  
1. In R:
```{r eval=FALSE}
script <- RCurl::getURL("https://raw.githubusercontent.com/pinskylab/pinskylab_methods/master/genomics/scripts/field_helpers.R", ssl.verifypeer = FALSE)
eval(parse(text = script))
```
## How to connect amphiprion Rstudio server to existing github repo
1. Follow githubs directions for setting up [SSH keys](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/).  
1. On the command line on amphiprion, type the SSH connection: 
```{bash eval=FALSE}
git clone git@github.com:user_name/repo_name.git
```
1. Open the the project file in the RStudio server for amphiprion in your web browser.  If you need the webaddress for the RStudio server, email Michelle. 

## How to use the command line to connect your current folder to an existing github repo.  
1. Open command line and navigate to the folder you want to connect to your github repo.  
1. On the command line, type:
```{bash eval=FALSE}
# set the new remote
git remote add origin url_of_github_repo

# verify that it worked
git remote -v 

# pull from the remote
git pull origin master

# put to the remote
git push origin master
```


# R
## How to install an older verison of a package than the one currently installed.   
Based on this [article](https://support.rstudio.com/hc/en-us/articles/219949047-Installing-older-versions-of-packages)  
```{r eval=FALSE}
devtools::install_version("ggplot2", version = "0.9.1", repos = "http://cran.us.r-project.org")

packageurl <- "https://github.com/thomasp85/gganimate/archive/v0.1.1.tar.gz"
install.packages(packageurl, repos=NULL, type="source")

packageloc <- "~/Downloads/gganimate-0.1.1/"
install.packages(packageloc, repos = NULL, type="source")
```




# Plots
## How to save an R plot as a pdf (when using ggplot) - 3 different ways
-  Save the last plot created into working directory.
```{r eval=FALSE}
ggplot2::ggsave("filename.pdf")
```

-  Save the object called plot into the working directory.  
```{r eval=FALSE}
ggplot2::ggsave(plot, "filename.pdf")
```

- Save the object called plot into the plots folder that is closest to your working directory.
```{r eval= FALSE}
ggplot2::ggsave(plot, here::here("plots", "filename.pdf"))
```
# Sequencing
## How to calculate sequencing coverage
The calculation is slightly different if you are doing RADSeq vs. whole genome sequencing.

For whole genome sequencing, number of base pairs is important.
To determine the number of runs needed for whole genome sequencing:

$${number~of~individuals~*~number~of~bp~in~the~genome~*~coverage~*~}\frac{1}{number~of~bp~per~read}~*~\frac{1}{number~of~reads~per~run}~=~number~of~runs$$

For ddRADSeq or other partial sequencing techniques where the sequencer always starts at the beginning of the contig of interest, number of reads replaces basepair numbers.  To determine the number of runs:

$${number~of~individuals~*~coverage~*~}\frac{1}{number~of~bp~per~read}~*~\frac{1}{number~of~reads~per~run}~=~number~of~runs$$
Also, [check out this pdf](https://github.com/pinskylab/pinskylab_methods/blob/master/genomics/coverage_calculation.pdf)  

# Video Conferencing
## google hangout for lab meetings
[HERE](https://hangouts.google.com/call/lgyfvg5ysbhhbhyjpdmj447mcqe)

## How to set up the ENR 145 TV for video conference.
-   Turn on the computer (on the shelf below the TV)
-   Switch the source using the input button on the remote to HDMI1 (the top port is broken as of March 2019, so you may have to move the cord to the bottom port)
-   Log in to the DEENR admin or Online user account. You can find the password on our Pinsky Lab Meeting Google Sheet.
-   Open chrome and navigate to the fish baste link. You can find the link on our Pinksy Lab Meeting Google Sheet. Other video conferencing software is also available.
-   Be sure to log out of Skype/Google/etc when you're done. 
-   Log out of the Windows user account.
-   Turn off computer and TV.
