# Pinskylab Organization

Repositories | Description
-----------------|---------------------------------
[Clownfish Tail Color](https://github.com/pinskylab/Clown_Fish_Tail_Color_Darrow) | **Anthony's analysis of tail color photos**
[Clownfish Persistence](https://github.com/pinskylab/Clownfish_persistence) | Allison
[Clownfish Genomics](https://github.com/pinskylab/genomics) | **Michelle's seq data processing and analysis**
[Clownfish succession](https://github.com/pinskylab/adriana) | Adriana's analysis of clownfish succession 
[Clownfish Group Structure](https://github.com/pinskylab/Clownfish_group_structure) | **Hannah's analysis of clownfish group structure** 
[Clownfish R package](https://github.com/pinskylab/clownfish) | An R package for accessing the database to work with clownfish 
[Clownfish Group Decisions](https://github.com/pinskylab/clownfish_group_decisions) | **Overarching decisions for Leyte clownfish projects** 
[Clownfish Field Code](https://github.com/pinskylab/field) | Scripts to be used in the field to enter and process data
[Clownfish Fecundity](https://github.com/pinskylab/YawdoszynClownfishFecundity) | **Adam's project on breeding females**
[Clownfish Transects](https://github.com/pinskylab/clownfishTransect) |  Cathrin's analysis of transect data 
[Clownfish Data Analysis](https://github.com/pinskylab/Clownfish_data_analysis) | **Allison's data analysis**
[Clownfish Dispersal](https://github.com/pinskylab/Dispersal) | Allison 
[Clownfish Philippines GIS](https://github.com/pinskylab/Phils_GIS_R) |  **Leyte map storage**  
[Regional Coral](https://github.com/pinskylab/Regional_Coral) | Lisa and Dan's regionally applied coral adaptation model 
[Fish Regression](https://github.com/pinskylab/Fish_regression_ABC) | **Jude's analysis of a size structured population** 
[Projection Uncertainty](https://github.com/pinskylab/project_uncertainty) | Jim's species distribution projection uncertainty 
[Hot Water](https://github.com/pinskylab/hotWater) | **Malin's comparision of climate impacts on land and sea** 
[Climate Velocity](https://github.com/pinskylab/speciesClimateVelocity) | Malin's calculations of species specific climate velocities 
[Project Velocity](https://github.com/pinskylab/project_velocity) | **Jim's projections of climate velocities to 2100** 
[Synchrony](https://github.com/pinskylab/cnhFishSynchrony) | Joyce's research on fish, fishing, and synchrony 
[Coral Adaptation](https://github.com/pinskylab/coral-adaptation) | **Lisa's coral adaptative capacity model** 
[Colonization Extinction](https://github.com/pinskylab/colonizationExtinction) |  Malin's colonization and extinction dynamics in NA continental shelf communities  [Portfolio Effect](https://github.com/pinskylab/Portfolio-Effect) | **Allison's portfolio effect ideas** 
[Coral Eco-Evo](https://github.com/pinskylab/ecoevo_coral) | <Ed's eco-evo models for coral adaptation to climate change. 
[Border Crossing](https://github.com/pinskylab/borderCrossing) | **Malin**
[Biome Variability](https://github.com/pinskylab/biomeVariability) |  Joyce's analysis of global variation in tempe/precip and relationship to plants
[Global Fish Genetics](https://github.com/pinskylab/globalFishGenetics) | **Malin's analysis of global marine patterns of fish genetic diversity** 
[Projected Ranges](https://github.com/pinskylab/proj_ranges) |  Becca
[Summer Flounder Connectivity](https://github.com/pinskylab/PADEconnectivity) | **Jennifer**
[Larval Analysis](https://github.com/pinskylab/NJNC-Larval-Analysis) | Jennifer
[Adult Summer Flounder](https://github.com/pinskylab/adultPADE) | **Jennifer**
[Freebayes vs. GATK](https://github.com/pinskylab/gatk-v-free) | Greg's analysis of SNP calling in freebayes vs. GATK
[How to submit proposal to CINAR](https://github.com/pinskylab/CINAR) | **Michelle's directions to prep for CINAR submission** 
[Leyte database maintenance](https://github.com/pinskylab/leyteBuildDB) | Michelle's repo that updates the Leyte database
[Pinskylab Methods](https://github.com/pinskylab/pinskylab_methods) | **you are here** 
[Lab Meetings](https://github.com/pinskylab/Lab_meetings) | A repo to hold documents related to Pinsky Lab meetings 
[Summer Flounder db maint](https://github.com/pinskylab/pade) | **Michelle's repo to update the summer flounder database** 
[Docs for Cross Spp Grant Sub](https://github.com/pinskylab/cross_spp_quote) |  Michelle's docs for cross species grant submission 
[Docs for Coastal SEES](https://github.com/pinskylab/coastalsees) | **Becca's docs for Coastal SEES**
[Ponyfish](https://github.com/pinskylab/zaman_ponyfish) | Marhuma's analysis of ponyfish
[Cod Evolution](https://github.com/pinskylab/codEvol)  |  **Malin's scripts for analysis of Atlantic cod genome sequences through time (Northeast Arctic and Canada)**

## Methods Repository navigation:

### [field](https://github.com/pinskylab/field)    

This folder has been replaced by the repository [field](https://github.com/pinskylab/field) for easier RStudio connection.  Maps are made using the repository [Phils_GIS_R](https://github.com/pinskylab/Phils_GIS_R).  The folder in this methods repository contains old scripts used in the Philippines by the Leyte team to input and process data sheets.  It also contains data from the temperature loggers and GPS data.

### [genomics](https://github.com/pinskylab/pinskylab_methods/tree/master/genomics)  

This folder contains subfolders and guides to help lab members working with DNA in the laboratory and also with bioinformatics.

### [labmgt](https://github.com/pinskylab/pinskylab_methods/tree/master/labmgt)    

This folder contains protocols for general lab management such as how to create backups.

### [cookbook.md](https://github.com/pinskylab/pinskylab_methods/blob/master/cookbook.md)  

This file gives tips and tricks for how Pinsky Lab Members have overcome obstacles related to data science.  The struggle is real!
