<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="exporter-version" content="Evernote Mac 6.6.1 (453373)"/><meta name="keywords" content="filtering, Pinsky"/><meta name="altitude" content="24.14390182495117"/><meta name="author" content="Michelle Stuart"/><meta name="created" content="2016-02-29 16:36:22 +0000"/><meta name="latitude" content="40.49265296497303"/><meta name="longitude" content="-74.42977155947344"/><meta name="source" content="desktop.mac"/><meta name="updated" content="2016-03-25 18:55:23 +0000"/><title>Filtering February 29, 2016 - SNP.DP10g95maf05.HWE - 1012 loci</title></head><body>
<div>Convert edited genepop to vcf (starting with the gene pop from which regenotyped and recaptured individuals have been removed)…can’t find anything on the web that will convert genepop to vcf - going to go back to our pre-genepop VCF and </div>
<ul>
<li><input checked="true" type="checkbox"/>filter to minor allele frequency of 0.2
<ul>
<li>The dDocent run from 11-12-15 was filtered twice, on Jan 22 to make a DP20 file  and Jan 25 to make a DP10 file.  </li>
<li>Our gene pop file comes from filtering the 11-Dec 2105 dDocent Total Raw SNPs on Jan 25 to make a DP10 file and is called  <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; background-color: #d6d6d6">SNP.DP10g95maf05.HWE.recode.vcf</span></li>
<li>Located in:  <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; background-color: #d6d6d6">/local/home/michelles/02-apcl-ddocent/APCL_analysis/11-12-2015/filtering_25-01-2016</span></li>
<li><span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; color: #4029d2; background-color: #d6d6d6">michelles 2016-02-29 11:54:49</span> <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; color: #009193; background-color: #d6d6d6">11-12-2015</span> <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; color: #00a100; background-color: #d6d6d6">$</span> <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; background-color: #d6d6d6">vcftools --vcf  filtering_25-01-2016/SNP.DP10g95maf05.HWE.recode.vcf --maf 0.2 --recode --recode-INFO-all --out filtering_25-01-2016/SNP.DP10g95maf2.HWE.recode.vcf</span></li>
<li><span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; background-color: #d6d6d6">After filtering, kept 990 out of 990 Individuals<br/>
After filtering, kept 1012 out of a possible 2620 Sites</span></li>
</ul>
</li>
<li><input checked="true" type="checkbox"/>convert to genepop
<ul>
<li><span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; color: #4029d2; background-color: #d6d6d6">michelles 2016-02-29 11:57:20</span> <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; color: #009193; background-color: #d6d6d6">filtering_25-01-2016</span> <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; color: #00a100; background-color: #d6d6d6">$</span> <span style="font: 14.0px 'Courier New'; font-variant-ligatures: no-common-ligatures; background-color: #d6d6d6"><a href="http://vcf2genepop.pl">vcf2genepop.pl</a> vcf=SNP.DP10g95maf2.HWE.recode.vcf.recode.vcf &gt; DP10g95maf2.genepop</span></li>
</ul>
</li>
<li><input checked="true" type="checkbox"/>fetch gene pop file</li>
<li><input checked="true" type="checkbox"/>add ‘pop’ to 3rd line of gene pop file</li>
<li><input checked="true" type="checkbox"/>rosetta stone samples in R - amended R script to handle any number of columns, previously was set to 2621, changed to ncol(largedf)-1</li>
<li><input checked="true" type="checkbox"/>remove regenotyped samples in R -
<ul>
<li><input checked="true" type="checkbox"/>have to run the regenotype script,</li>
<li><input checked="true" type="checkbox"/>then evaluate the csv by determining which to keep, (used the conservative logic that even if there were fewer loci, if there was no labeling error it was better to keep)</li>
<li><input checked="true" type="checkbox"/>then run evaluated script.</li>
</ul>
</li>
<li><input checked="true" type="checkbox"/>remove recaptured samples</li>
<li><input checked="true" type="checkbox"/>run through cervus</li>
</ul>
</body></html>