Files get filtered for:

- SNPS genotyped in % individuals (g)
- minor allele count (mac)
- quality (q)
- minimum depth (dp)
- individuals that did not seq well (lm)
- minor allele frequency (maf)
- HWE
- define indels (prim)
- Remove indels (SNP)

Work in the directory where SNPs were called.

1. Keep variants that have been successfully genotyped in 50% of individuals, have a minimum quality score of 30, and a minor allele count of at least 3.  This will get rid of a lot of data and make the next filtering steps run faster. Takes about 2 hours to run on 1.5 million loci.

 `vcftools --vcf TotalRawSNPs.vcf --geno 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out seq17_03g50mac3q30`

2. Next filter for minimum depth for a genotype call.  This filter examines the value in the DP column for each genotype and marks any genotype with a value less than 3 as "missing".  Takes about 2.5 hours.

 `vcftools --vcf seq17_03g50mac3q30.recode.vcf --minDP 3 --recode --recode-INFO-all --out seq17_03g50mac3q30dp3`

3. Get rid of individuals that did not sequence well by removing any individuals who have more than X % missing data.  Takes about 2 hours.

 `filter_missing_ind.sh seq17_03g50mac3q30dp3.recode.vcf  seq17_03g50mac3q30dp3lm`
 
 Look at the histogram and make sure that using the default is the best choice.
  
4. Restrict the data to variants called in a high percentage of individuals (95%) and filter by mean depth of genotypes (10). For identity analysis and parentage, use a more strict cutoff, reducing the number of loci to alleles found in a larger proportion of the population.  Takes about 10 minutes.

 `vcftools --vcf seq17_03g50mac3q30dp3lm.recode.vcf --geno 0.95 --min-meanDP 12 --maf 0.2 --recode --recode-INFO-all --out seq17_03g95maf2q30dp12`

6. Begin filtering with FreeBayes designed output - run script dDocent filters.  Takes about 5 minutes.

 `dDocent_filters seq17_03g95maf2q30dp20lmp10.recode.vcf seq17_03g95maf2q30dp20lmp10`

 *Is this from a mixture of SE and PE libraries? Enter yes or no.*

 no

 *If distrubtion looks normal, a 1.645 sigma cutoff (~90% of the data) would be 18.9923411111.  The 95% cutoff would be 17. Would you like to use a different maximum mean depth cutoff than 17, yes or no*

 no

8. Convert file to SNPs only and indicate SNP and INDELs. Takes about 5 minutes.

 `vcfallelicprimitives seq17_03g95maf2q30dp20lm.HWE.recode.vcf --keep-info --keep-geno > seq17_03g95maf2q30dp20lm.HWE.prim.vcf`

9. Remove INDELs - takes about 30 seconds.

 `vcftools --vcf seq17_03g95maf2q30dp20lm.HWE.prim.vcf --remove-indels --recode --recode-INFO-all --out SNP.seq17_03g95maf2q30dp20lm.HWE`

11. Convert to genepop

 `vcf2genepop.pl vcf=SNP.seq17_03g95maf2q30dp20lm.HWE.filtered.vcf > seq17_03g95maf2q30dp20lm.gen`

12. Fetch genepop file

13. Add ‘pop’ to 3rd line of genepop file

14. Run the [process_genepop.R](https://github.com/stuartmichelle/Genetics/blob/master/code/process_genepop.R) script to remove regenotyped samples and samples that are known to have issues.

15. Run through identity analysis in cervus to find recaptures

16. Run the [identity_analysis.R]() script to remove recaptured samples so that each individual is only represented once in the data set that is analyzed for parentage.

[Return to analysis protocol](./0.hiseq_ddocent.md)
